packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -456,52 +457,60 @@ class _BottomNavigationTile extends StatelessWidget {
         break;
     }
 
-    return Expanded(
-      flex: size,
+    Widget result = MergeSemantics(
       child: Semantics(
         container: true,
         selected: selected,
         child: Stack(
           children: <Widget>[
-            InkResponse(
-              onTap: onTap,
-              mouseCursor: mouseCursor,
-              child: Padding(
-                padding: EdgeInsets.only(top: topPadding, bottom: bottomPadding),
-                child: Column(
-                  crossAxisAlignment: CrossAxisAlignment.center,
-                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
-                  mainAxisSize: MainAxisSize.min,
-                  children: <Widget>[
-                    _TileIcon(
-                      colorTween: colorTween,
-                      animation: animation,
-                      iconSize: iconSize,
-                      selected: selected,
-                      item: item,
-                      selectedIconTheme: selectedIconTheme ?? bottomTheme.selectedIconTheme,
-                      unselectedIconTheme: unselectedIconTheme ?? bottomTheme.unselectedIconTheme,
-                    ),
-                    _Label(
-                      colorTween: colorTween,
-                      animation: animation,
-                      item: item,
-                      selectedLabelStyle: selectedLabelStyle ?? bottomTheme.selectedLabelStyle,
-                      unselectedLabelStyle: unselectedLabelStyle ?? bottomTheme.unselectedLabelStyle,
-                      showSelectedLabels: showSelectedLabels ?? bottomTheme.showUnselectedLabels,
-                      showUnselectedLabels: showUnselectedLabels ?? bottomTheme.showUnselectedLabels,
-                    ),
-                  ],
-                ),
-              ),
-            ),
             Semantics(
```

Comment:
Add `container: true` to this Semantics widget and move it back to after the `Tooltip` widget in the Stack. That should fix the issue you mentioned on Chat.

(Basically replace lines 466-513 in this PR with the content of this gist: https://gist.github.com/goderbauer/8913e43d987e76b7b89451dd2ac89089)

Reply:
As explanation: It looks this is a quirk of the semantics compiler. Since the `indexLabel` was not in a container, its semantics information were implicitly merged to the highest node possible in the tree (which here happens to be the `MergeSemantics` widget). Due to this, before the force-merge triggered by the MergeSemantics widget happens, the SemanticsNode of the MergeSemantics widget already has a label of `indexLabel`. During the force-merge, the labels of the child SemanticsNodes (here: only the one of the Tooltip) are added to the end of that label. Hence, no matter how you order the children in the stack you get the incorrect merged label.

If you now make the `indexLabel` a container itself, it will own its own semantics node. The MergeSemantics widget now has two semanticsNode children and they get force-merged in the order the children have in the tree.

Hope that's somewhat clear. If not, let me know.

Reply:
This worked!! Thank you for spotting that. Very tricky.